p2p-tests/star-topology.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+# Copyright © 2021 The Radicle Upstream Contributors
+#
+# This file is part of radicle-upstream, distributed under the GPLv3
+# with Radicle Linking Exception. For full terms see the included
+# LICENSE file.
+
+set -euo pipefail
+
+function usage {
+  cat <<EOF
+Usage: sudo $(basename "${BASH_SOURCE[0]}") start | stop
+
+This script sets up a star network topology.
+It only works on Linux and _has to be run as root_.
+
+EOF
+  exit
+}
+
+function setup_network {
+  echo "setting up network: maintainer <---> seed <---> contributor"
+
+  ip netns add maintainer
+  ip netns add seed
+  ip netns add contributor
+
+  ip link add veth0 type veth peer name veth1
```

Comment:
We should use a naming scheme that scopes all the network interfaces and also relates the network interfaces better to the purpose. E.g. `upstream-test-maintainer-host`, `upstream-test-maintainer-bridge`, `upstream-test-bridge`, etc.

Reply:
Yeah, I agree. I just wonder what's the best scheme for the `veth` pairs. Because it has two ends, so should it be `upstream-test-maintainer-seed-A-veth` and `upstream-test-maintainer-seed-B-veth`? Or something different, kinda gets long to type it out.

Reply:
For the veth pairs I’d suggest `upstream-test-${peer_name}-host` for the end `peer_name` binds to and `upstream-test-${peer_name}-bridge` for the end that is connected to the bridge. We could also do `upstream-test-${peer_name}.host` and `upstream-test-${peer_name}.bridge` to make it clearer that they are the two ends.

In the naming scheme that you proposed I don’t quite get what the `seed-X` part is telling us. Is it indicating the IP layer star topology with the seed at the center? If so this is probably confusing because the interfaces determine the MAC layer topology which is a star topology with the bridge at the center.

Reply:
Here's a diagram of the current topology, so it's easier to discuss:

![IMG_0091](https://user-images.githubusercontent.com/158411/143463172-b6f245c2-7094-449d-8fb8-a9bcab8b4ccf.jpg)

Reply:
9718c5bd95ee03357efbad307a1da9343aec7eb0

Reply:
Ugh, there's a 15 char limit on ifnames 🤦